config/dynamic.c

```diff
@@ -16,11 +16,546 @@
+
+	/* The lua state used to handle the dynamic configuration files. */
+	lua_State *lua_state;
+
+	len = sizeof(client_addr);
+	conn_fd = accept(server_socket_fd,
+		(struct sockaddr *)&client_addr, &len);
+	if (conn_fd < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to accept a new connection - (%s)\n",
+			strerror(errno));
+		return;
+	}
+
+	if (unlikely(client_addr.sun_family != AF_UNIX)) {
+		RTE_LOG(WARNING, GATEKEEPER,
+			"Unexpected condition: unknown client type %d at %s\n",
+			client_addr.sun_family, __func__);
+		goto close_fd;
+	}
+
+	/*
+	 * XXX The request must be received under a specified timeout,
+	 * or the request is aborted. This code needs further test.
+	 */
+	ret = setsockopt(conn_fd, SOL_SOCKET, SO_RCVTIMEO,
+		(const char*)&dy_conf->rcv_time_out, sizeof(struct timeval));
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to call setsockopt(SO_RCVTIMEO) - (%s)\n",
+			strerror(errno));
+		goto close_fd;
+	}
+
+	rcv_buff_size = MSG_MAX_LEN;
+	ret = setsockopt(conn_fd, SOL_SOCKET,
+		SO_RCVBUF, &rcv_buff_size, sizeof(rcv_buff_size));
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to call setsockopt(SO_RCVBUF) with size = %d - (%s)\n",
+			rcv_buff_size, strerror(errno));
+		goto close_fd;
+	}
+
+	lua_state = luaL_newstate();
+	if (lua_state == NULL) {
+		RTE_LOG(ERR, LUA,
+			"dyn_cfg: failed to create new Lua state!\n");
+		goto close_fd;
+	}
+
+	/* Set up the Lua state while there is a connection. */
+	ret = setup_dy_lua(lua_state);
+	if (ret < 0) {
+		RTE_LOG(ERR, LUA,
+			"dyn_cfg: Failed to set up the lua state\n");
+		goto close_lua;
+	}
+
+	while (1) {
+		ret = process_single_cmd(conn_fd, lua_state);
+		if (ret != 0)
+			break;
+	}
+
+close_lua:
+	lua_close(lua_state);
+
+close_fd:
+	ret = close(conn_fd);
+	if (ret < 0) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"dyn_cfg: Failed to close the connection socket - (%s)\n",
+			strerror(errno));
+	}
+}
+
+static int
+dyn_cfg_proc(void *arg)
+{
+	int ret = 0;
+	struct dynamic_config *dy_conf = arg;
+	uint32_t lcore = dy_conf->lcore_id;
+
+	RTE_LOG(NOTICE, GATEKEEPER,
+		"dyn_cfg: the Dynamic Config block is running at lcore = %u\n", lcore);
+
+	while (likely(!exiting)) {
+		fd_set fds;
+		struct timeval stv;
+ 
+		FD_ZERO(&fds);
+		FD_SET(dy_conf->sock_fd, &fds);
+ 
+		/*
+		 * 10000 usecs' timeout for the select() function.
+		 * This parameter can prevent the select() function
+		 * from blocking forever. So, the whole program can
+		 * exit when receiving a quitting signal.
+		 */
+		stv.tv_sec = 0;
+		stv.tv_usec = 10000;
+
+		ret = select(dy_conf->sock_fd + 1, &fds, NULL, NULL, &stv);
+		if (ret < 0 && errno != EINTR) {
+			RTE_LOG(ERR, GATEKEEPER,
+				"dyn_cfg: Failed to call the select() function - (%s)\n",
+				strerror(errno));
+			break;
+		} else if (likely(ret <= 0)) {
+			if (unlikely(ret < 0)) {
+				RTE_LOG(ERR, GATEKEEPER,
+					"dyn_cfg: Failed to call the select() function - (%s)\n",
+					strerror(errno));
+			}
```

Comment:
I know that @cjdoucette asked for this log message, but it's not needed. If `ret < 0` here, `errno` must be `EINTR`, and this condition is not an error. You can replace the whole `if (unlikely(ret < 0))` statement with `RTE_VERIFY(errno == EINTR);`